src/deluge/drivers/pic/pic.h

```diff
@@ -0,0 +1,300 @@
+	 * @param indicator An array of brightness values for each LED
+	 */
+	static void setGoldKnobIndicator(bool which, const std::array<uint8_t, kNumGoldKnobIndicatorLEDs>& indicator) {
+		Message knob = which ? Message::SET_GOLD_KNOB_0_INDICATORS : Message::SET_GOLD_KNOB_1_INDICATORS;
+		send(knob, indicator);
+	}
+
+	static void setLEDOff(size_t idx) { send(util::to_underlying(Message::SET_LED_OFF) + idx); }
+	static void setLEDOn(size_t idx) { send(util::to_underlying(Message::SET_LED_ON) + idx); }
+
+	/**
+	 * @brief Request that the PIC resend all button states
+	 */
+	static void resendButtonStates() { send(Message::RESEND_BUTTON_STATES); }
+
+	static void setMinInterruptInterval(uint8_t time_ms) { send(Message::SET_MIN_INTERRUPT_INTERVAL, time_ms); }
+	static void setFlashLength(uint8_t time_ms) { send(Message::SET_FLASH_LENGTH, time_ms); }
+
+	static void setUARTSpeed() {
+		uint8_t speed = 4000000.0f / kUartFullSpeedPadsHz - 0.5f; // Speed is 4MHz / (x + 1)
+		send(Message::SET_UART_SPEED, speed);
+	}
+
+	static void flashPad(size_t idx) { send(util::to_underlying(Message::SET_PAD_FLASHING) + idx); }
+
+	/**
+	 * @brief Flash a pad using the PIC's built-in timer and colour system
+	 *
+	 * @param idx The pad to flash
+	 * @param colour_idx The colour "index" of colours the PIC knows
+	 */
+	static void flashPadWithColourIdx(size_t idx, int32_t colour_idx) {
+		send(util::to_underlying(Message::SET_FLASH_COLOR) + colour_idx);
+		flashPad(idx);
+	}
+
+	static void update7SEG(const std::array<uint8_t, kNumericDisplayLength>& display) {
+		send(Message::UPDATE_SEVEN_SEGMENT_DISPLAY, display);
+	}
+
+	static void enableOLED() { send(Message::ENABLE_OLED); }
+	static void selectOLED() { send(Message::SELECT_OLED); }
+	static void deselectOLED() { send(Message::DESELECT_OLED); }
+	static void setDCLow() { send(Message::SET_DC_LOW); }
+	static void setDCHigh() { send(Message::SET_DC_HIGH); }
+
+	static void requestFirmwareVersion() { send(Message::REQUEST_FIRMWARE_VERSION); }
+
+	static void sendColour(const Colour& colour) { send(colour); }
+
+	static void setRefreshTime(uint8_t time_ms) { send(Message::SET_REFRESH_TIME, time_ms); }
+	static void setDimmerInterval(uint8_t interval) { send(Message::SET_DIMMER_INTERVAL, interval); }
+
+	/**
+	 * @brief This is used to make smooth scrolling on the horizontal axis, by filling up an 8-pixel framebuffer on the PIC
+	 *        that it can then provide an animation to smear across the axis to give a sensation of motion
+	 *        See (scrolling between clip pages)
+	 *
+	 * @param idx the row to set
+	 */
+	static void sendScrollRow(size_t idx, Colour colour) {
+		send(util::to_underlying(Message::SET_SCROLL_ROW) + idx);
+		send(colour);
+	}
+
+	static void setupHorizontalScroll(uint8_t bitflags) {
+		send(util::to_underlying(Message::SET_SCROLL_LEFT) + bitflags);
+	}
+
+	static void setupVerticalScroll(bool direction, const std::array<Colour, kDisplayWidth + kSideBarWidth>& colours) {
```

Comment:
IIUC this is more like `doVerticalScroll` as this command performs the actual scroll.

Reply:
Fixed.